framework/src/kernels/ArrayTimeDerivative.C

```diff
@@ -0,0 +1,95 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ArrayTimeDerivative.h"
+
+registerMooseObject("MooseApp", ArrayTimeDerivative);
+
+template <>
+InputParameters
+validParams<ArrayTimeDerivative>()
+{
+  InputParameters params = validParams<ArrayTimeKernel>();
+  params.addClassDescription("Array time derivative operator with the weak form of $(\\psi_i, "
+                             "\\frac{\\partial u_h}{\\partial t})$.");
+  params.addParam<MaterialPropertyName>("time_derivative_coefficient",
+                                        "The name of the time derivative coefficient. "
+                                        "Can be scalar, vector, or matrix");
+  return params;
+}
+
+ArrayTimeDerivative::ArrayTimeDerivative(const InputParameters & parameters)
+  : ArrayTimeKernel(parameters)
+{
+  if (hasMaterialProperty<Real>("time_derivative_coefficient"))
+  {
+    _coeff_type = 0;
+    _coeff = &getMaterialProperty<Real>("time_derivative_coefficient");
+  }
+  else if (hasMaterialProperty<RealEigenVector>("time_derivative_coefficient"))
+  {
+    _coeff_type = 1;
+    _coeff_array = &getMaterialProperty<RealEigenVector>("time_derivative_coefficient");
+  }
+  else if (hasMaterialProperty<RealEigenMatrix>("time_derivative_coefficient"))
+  {
+    _coeff_type = 2;
+    _coeff_2d_array = &getMaterialProperty<RealEigenMatrix>("time_derivative_coefficient");
+  }
+  else
+  {
+    MaterialPropertyName mat = getParam<MaterialPropertyName>("time_derivative_coefficient");
+    mooseError("Property " + mat + " is of unsupported type for ArrayTimeDerivative");
+  }
+}
+
+RealEigenVector
+ArrayTimeDerivative::computeQpResidual()
+{
+  if (_coeff_type == 0)
```

Comment:
These could probably just check `if (_coeff/_coeff_array/_coeff_2d_array)` since only one will be non-null.  Then we don't need _coeff_type at all.

Reply:
I removed `_coeff_type` per your suggestion. But I kept getting random crashes, I guess because the null ones were never initiated. So I switched to a ternary operator to assure the unused ones are initiated to `nullptr`.

Reply:
Yes, as a general rule, we need to make sure every member variable that is a POD type (e.g. pointers, ints, etc.) is explicitly initialized in the constructor.  What you did was fine.  Another fine option would be to just set all of them to nullptr in the initialization list and leave the if checking code in the constructor body like before.